core/src/com/unciv/ui/cityscreen/CityScreen.kt

```diff
@@ -133,6 +136,21 @@ class CityScreen(
         }
     }
 
+    private fun playCitySound() {
+        when (city.civInfo.getEra().eraNumber) {
+            0 -> SoundPlayer.play(UncivSound.CityAncient)
+            1 -> SoundPlayer.play(UncivSound.CityClassical)
+            2 -> SoundPlayer.play(UncivSound.CityMedieval)
+            3 -> SoundPlayer.play(UncivSound.CityRenaissance)
+            4 -> SoundPlayer.play(UncivSound.CityIndustrial)
+            5 -> SoundPlayer.play(UncivSound.CityModern)
+            6 -> SoundPlayer.play(UncivSound.Click) //atomic
+            7 -> SoundPlayer.play(UncivSound.Click) //information
+            8 -> SoundPlayer.play(UncivSound.Click) //future
+            else -> SoundPlayer.play(UncivSound.Silent)
```

Comment:
Inherently unmoddable, why not check if a sound file exists with the name of the era, if yes play it, if no then click?

Reply:
can't quite tell how to do this. If I replace `UncivSound.CityAncient`  with `UncivSound("cityAncient")` will it work ?

Reply:
Can you just play a sound with something like `UncivSound("city" + city.civInfo.getEra().name)` with a catch for if the sound file doesn't exist (maybe play a default sound like `UncivSound.Click`)?

Reply:
> UncivSound.Click

I only used that to test that its actually playing a sound. The city name button already plays that sound so the default would be Sound.Silent

> UncivSound("city" + city.civInfo.getEra().name)

just did this. But I used the era number instead of the name as the name was giving `Ancient Era` and not just `Ancient` 
But would this be moddable?

Reply:
There was a time long long ago when UncivSound was an enum and the attackSound present in json but a dead fish doing nothing. To avoid the mutable enum instance debacle that _was_ happily done elsewhere later, I made it a class allowing runtime-instantiation but keeping source compatibility for the existing "enum constants" which are now explicitly pre-created singletons (enum "constants" are too under the hood).... Just so you can understand better. A new "predefined" might be good if you expect it to be used in dozens of places throughout the code, otherwise don't hesitate to just use the constructor as shown.

> file not found ex. I guess its already handled

Read the sound code. Looks a whopper but most is just caching (Gdx dox on Sounds say "dispose!", reloading _can_ be heard, so the cache has a cleanup, everybody happy). To allow mod sounds it needs to do a folder search, so of course the individual "not found" just moves on - what I did when all folders exhausted I've forgotten - hey I can't check, the class is _gone_ :roll_eyes: Ah there it is, another helpful rename - yup, get does debuglog, cache the "not found" fact, return null; and play does get ?: return. You could extend the "API" of SoundPlayer with a `play(): Boolean` return or with a `isCached(sound: UncivSound) = sound in soundMap` helper or a `play(, fallback: UncivSound? = null)` parameter or probably a few other ways if you need predictable fallbacks.